### PR TITLE
[sec_scan][20] add `ReportSecrets` forwarder to proxy's gRPC insecure server

### PR DESCRIPTION
**Pull request by tigrato**
This PR introduces a `ReportSecrets` forwarder that routes requests from the Proxy server to the Auth server. The objective is to enable clients to access the proxy's insecure gRPC server (without credentials), and have the proxy forward these requests to the AuthServer on their behalf. This is necessary because the client lacks valid credentials and cannot reach the auth server through the reversetunnel when the cluster operates in `separate` mode.

This PR is part of https://github.com/gravitational/access-graph/issues/637.

### CODE REVIEW

File: lib/secretsscanner/client/client.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	// Insecure trusts the certificates from the Auth Server or Proxy during registration without verification.
+	Insecure bool
+	// Log is the logger.
+	Log *slog.Logger
+}
+
+// NewSecretsScannerServiceClient creates a new SecretsScannerServiceClient that connects to the proxy
+// gRPC server that does not require authentication (credentialless) to report secrets found during scanning.
+func NewSecretsScannerServiceClient(ctx context.Context, cfg ClientConfig) (Client, error) {
+	if cfg.ProxyServer == "" {
```
Comment (smallinsky) [inline lib/secretsscanner/client/client.go:81]: > gRPC server that does not require authentication (credentialless) to report secrets found during scanning.

Thy is the reason why the `ReportSecrets(ctx context.Context, opts ...grpc.CallOption) (accessgraphsecretsv1pb.SecretsScannerService_ReportSecretsClient, error)` doesn't need any  auth ? 


Anyone will can connect and report arbitrary data - How we will protect againts DDOS where the attacker can sent artificial data to drain our backend throughput and drastically increase database size ?

File: lib/secretsscanner/client/client.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+func (c *secretsSvcClient) Close() error {
+	return c.conn.Close()
+}
+
+// proxyConn attempts to connect to the proxy insecure grpc server.
+// The Proxy's TLS cert will be verified using the host's root CA pool
+// (PKI) unless the --insecure flag was passed.
+func proxyConn(
```
Comment (smallinsky) [inline lib/secretsscanner/client/client.go:114]: Seems to be duplicated with https://github.com/gravitational/teleport/blob/master/lib/auth/join/join.go#L407 

Can we extract common function ?

File: lib/secretsscanner/proxy/proxy.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+// from the proxy to the Auth's secret service.
+type Service struct {
+	accessgraphsecretsv1pb.UnimplementedSecretsScannerServiceServer
+	// authClient is the client to the Auth service.
+	authClient AuthClient
+
+	log *slog.Logger
+}
+
+func (s *Service) ReportSecrets(client accessgraphsecretsv1pb.SecretsScannerService_ReportSecretsServer) error {
```
Comment (smallinsky) [inline lib/secretsscanner/proxy/proxy.go:72]: nit: godocs

File: lib/secretsscanner/proxy/proxy.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	"io"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+
+	accessgraphsecretsv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/accessgraph/v1"
+)
+
+// AuthClient is a subset of the full Auth API that must be connected
+type AuthClient interface {
```
Comment (smallinsky) [inline lib/secretsscanner/proxy/proxy.go:35]: nit 
```suggestion
type authClient interface {
```

File: lib/secretsscanner/client/client.go (hunk 1 above)
Comment (smallinsky) [reply lib/secretsscanner/client/client.go:81]: Make sense. 

The auth is done by  auth ReportSecrets handleer:  https://github.com/gravitational/teleport.e/pull/4649/files#diff-346834ba5299731121f1d806e1b0ca2b2046c2ccc0157bcb41bcd2dff871e929R98  I have missed the reference to the server part.

**Review by smallinsky — APPROVED** [review]

**Review by vapopov — APPROVED** [review]

**Pull request merged**